enginetest/queries/script_queries.go

```diff
@@ -3749,6 +3749,58 @@ var PreparedScriptTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "prepare using user vars",
+		SetUpScript: []string{
+			"create table t (i int primary key);",
+			"insert into t values (0), (1), (2);",
+			"set @a = 'select * from t order by i'",
+			"set @b = concat('select 1',' + 1')",
```

Comment:
maybe some examples with bind variables in the statement, and referencing user variables from a statement string. just combinations of variable nesting/composition to make sure all the related code is generalized